subscription.tf

```diff
@@ -1,12 +1,51 @@
 locals {
   enable_subscription = length(var.subscribed_log_group_matches) + length(var.subscribed_log_group_excludes) > 0
+  observe_log_groups  = [aws_cloudwatch_log_group.group.name, "/aws/lambda/${var.name}"]
```

Comment:
Implicit dependency on Lambda here, likely to fail on initial creation. Reverse engineering of IDs is always a red flag for this reason. 

The general approach to this is to declare the CW logs group with Terraform, reverse engineering the name there and only there. AWS will use that existing log group, at least in my experience with other services with a similar problem like EKS. 

But this is a breaking change, as it requires either replacement of the Lambda (new name) or using `terraform import` for the log group. 

Doable if you consider this from the start but a chore if you're trying to migrate.